### ISSUE TITLE
GeanyLatex and "Make": Shortcuts become disabled

### ISSUE THREAD

**Issue by BenWiederhake** [geany/geany-plugins#282]
This one is hard to reproduce (sorry), and I have no idea what causes this or what is happening.

Steps to reproduce *sometimes*:

0. Install and enable GeanyLaTeX (in case it is relevant: I have only the "auto-completion" activated)
1. Create and open: A latex file and a Makefile, in the same directory (I use the makefile to compile the latex document)
2. Edit the latex file over a long time, intermittently calling "make" via the shortcut "Shift+F9" (as found in the menu)

Actual behaviour: Sometimes the "Make" menu item becomes and stays disabled. This also disables the keyboard shortcut, so Shift+F9 does not do anything anymore. Switching between geany-tabs or re-opening the file does not help. The only way to work around this is to restart Geany.

Expected behaviour: Shift+F9 and the "Make" menu item stay enabled for the whole time.

Versions: Geany 1.25 "Veed" // "(compiled on 2015-07-12 or later)"

Sorry that I can't provide a more reliable way of reproducing the bug. I have already played with different things, and have, so far, only come to the conclusion that disabling the GeanyLaTeX pluging avoids this bug. However, this is not a guarantee, since I "only" tried that for a week. Otherwise I run into that bug roughly once every two weeks, so this experiment isn't conclusive.

I continue using Geany, so if anyone can give me a better "test", or instructions on what to look for via gdb while this is happening, I'd gladly do that.

**Comment by frlan** [comment]
Is only the Make-menu-item effected by this? Do you see any output in Debug (Help->Debug) or while starting Geany from prompt?

**Comment by BenWiederhake** [comment]
All "Make" menu-items are affected when the bug is triggered.

I will look at Help->Debug the next time this happens. For the record, in the current run (where the bug hasn't triggered yet), I see several "INFO" lines, so Help->Debug works and there do not seem to be any warning or errors so far.

I always start Geany from a command-line, and never see any warnings or errors. I am not sure which application causes the following warning (a failing assertion should be more than a warning, shouldn't it?), so it may or may not be Geany.

    (process:31772): GLib-CRITICAL **: g_slice_set_config: assertion 'sys_page_size == 0' failed

(A lot of applications produce this "warning", so I may have a bad setup, even though it's a standard Debian testing.)

**Comment by BenWiederhake** [comment]
I believe I found the core issue: geany really doesn't like Makefiles that spawn daemons, e.g. contain a target that executes `program &` or similar.

Workaround: as soon as the called `program` returns, geany can `make` again.  In my case it's `evince`, which exits immediately if there already is another instance running (which then reloads the PDF in question), so this is very easy for me.

However, I believe that Geany should be able to `make` again as soon as `make` returns, so I do consider this a bug.

**Comment by b4n** [comment]
> However, I believe that Geany should be able to `make` again as soon as `make` returns, so I do consider this a bug.

I don't think so.  I'm not so much knowledgeable about that, but AFAIK for the background jobs not to block Geany's spawning, it would require for Geany to inherit those children from the process it spawned, as they didn't deamonize themselves and need a parent.
I don't know what Geany could even do (portably, nonetheless) to achieve that, but I'm not quite sure it makes much sense to support that.

I understand your use case, but it seems fairly hacky to put that in the Makefile.  I myself do something similar with Lilypond files, but have a separate (run) command to show the generated PDF, which doesn't have the problem you're facing.
